### PR TITLE
Textinput

### PR DESCRIPTION
**Pull request by jzhang300**
* Text input has underline that matches the text width

### CODE REVIEW

File: example/src/components/TextInputExamples.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import TextInput from '../../../src/components/TextInput';
+import Code from '../../../src/components/Code';
+// import Icon from '../../../src/components/Icon';
```
Comment (kognate) [inline example/src/components/TextInputExamples.js]: if this is commented out we should probably just remove it.

File: src/components/Modal.js [hunk 2]
```diff
@@ -14,8 +14,10 @@ export default React.createClass({
     onEnter: React.PropTypes.func,
   },
 
-  defaultProps: {
-    isOpen: false,
+  getDefaultProps() {
```
Comment (kognate) [inline src/components/Modal.js:17]: isn't this from #23  ?

File: src/components/TextInput.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    };
+  },
+
+  onInput(e) {
+    const dummy = e.target.nextSibling;
+    // this is a hack for when you type space,
+    // input text gets shifted because of mismatched width between
+    // dummy and real input.  Char 'r' is used because it has the same
+    // length as a char ' ' in order for the width of the space to count.
+    const text = e.target.value.replace(/\s/g, 'r');
```
Comment (kognate) [inline src/components/TextInput.js:38]: \s will also match a tab, so the calc isn't right for tabbed text.  `"\t ".match(/\s/)`

**Review by kognate — CHANGES_REQUESTED** [review]
Not big comments, except an answer to the `\s` question.

File: src/components/TextInput.js (hunk 3 above)
Comment (jzhang300) [reply src/components/TextInput.js:38]: Would the user be able to type in tabs?  Because tabbing in the browser will focus on next element.

Comment (kognate) [reply src/components/TextInput.js:38]: copy/paste can include tabs

File: src/components/Modal.js (hunk 2 above)
Comment (jzhang300) [reply src/components/Modal.js:17]: eh, it is... I thought it wouldn't appear if I made the second pr in another branch.

File: src/components/TextInput.js (hunk 3 above)
Comment (jzhang300) [reply src/components/TextInput.js:38]: mm good point

**Review by kognate — APPROVED** [review]
LGTM

**Pull request merged**